jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -371,77 +371,79 @@ public CRIUSupport registerRestoreEnvFile(Path envFile) {
 
 	@SuppressWarnings("restriction")
 	private void registerRestoreEnvVariables() {
-		J9InternalCheckpointHookAPI.registerPostRestoreHook(RESTORE_ENVIRONMENT_VARIABLES_PRIORITY,
-				"Restore environment variables via env file: " + envFile, () -> { //$NON-NLS-1$
-					if (!Files.exists(this.envFile)) {
-						throw throwSetEnvException(new IllegalArgumentException(
-								"Restore environment variable file " + envFile + " does not exist."));
-					}
-
-					String file = envFile.toAbsolutePath().toString();
-
-					try (BufferedReader envFileReader = new BufferedReader(new FileReader(file))) {
-						/* [IF JAVA_SPEC_VERSION < 17] */
-						SecurityManager manager = System.getSecurityManager();
-						if (manager != null) {
-							manager.checkRead(file);
+		if (this.envFile != null) {
```

Comment:
Personally, I'd be tempted to do:
```java
if (this.envFile == null) {
    return;
}
```
and leave the rest of the code at the previous indent level.  We don't enforce single-return point in java code.